src/client/components/UsersListComponent/UsersList.js

```diff
@@ -1,21 +1,36 @@
-import React from 'react';
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable camelcase */
+/* eslint-disable @typescript-eslint/camelcase */
+import React, { useContext } from 'react';
+import { useHistory } from 'react-router-dom';
 import PropTypes from 'prop-types';
+import { UserContext } from '../../context/userContext';
 import './UsersListStyle.css';
+import { onStartChat } from '../../containers/UserList/onStartChat';
 
 export default function Userslist({ users }) {
+  const history = useHistory();
+  const user = useContext(UserContext);
+  const createConversation = onStartChat(history);
```

Comment:
All of this should go to a `container component` and in there you use history, context, and handlers.
Components should preferably only contain HTML/CSS and use props to communicate with the parent container.
And components should never import anything from containers, only the other way around.

Reply:
I tried this, but I had problems 'React Hook  is called in function  which is neither a React function component or a custom React Hook function'. but now I resolved it. Mistake was here 'when you are using React Hooks, your functional component's name MUST start with a capital letter'. only should change one small ting - capital letter :(